engine/sync.go

```diff
@@ -262,6 +254,18 @@ func newSyncServer(e *Engine) *syncServer {
 	}
 }
 
+// newSession allocates a new session ID and starts managing the session with
+// the provided node.
+func (s *syncServer) newSession(node net.IP) *syncSession {
```

Comment:
Function name wise, it's kind of difficult to differentiate this one from `newSyncSession`. Both are returning a new pointer to syncSession struct.

Probably rename this one as `registerNewSession`? Just my 2 cents.

Reply:
Since `newSyncSession` isn't used anywhere else, I tried inlining it. WDYT?

Reply:
Looks good. Thanks!